File: 이진영/Hash/전화번호 목록.js
function solution(phone_book) {
    let set = new Set(phone_book);
    
    for (let phone of phone_book) {
        for (let i = 0; i < phone.length; i++) {
            if (set.has(phone.slice(0, i))) return false;
        }
    }
    return true;
}

File: 김효선/Hash/day08/전화번호 목록.py
def solution(phone_book):
    phone_dict = {}

    for phone in phone_book:
        phone_dict[phone] = 1
    
    for phone in phone_book:
        x = ""
        for char in phone:
            x += char
            if x in phone_dict and x != phone:
                return False
    
    return True